pkg/v1/remote/write.go

```diff
@@ -41,17 +41,91 @@ type manifest interface {
 
 // Write pushes the provided img to the specified image reference.
 func Write(ref name.Reference, img v1.Image, options ...Option) error {
-	ls, err := img.Layers()
+	return WriteMulti([]name.Reference{ref}, img, options...)
+}
+
+// WriteMulti pushes the provided img to multiple specified image references.
+func WriteMulti(refs []name.Reference, img v1.Image, options ...Option) error {
+	// dedupe registries
```

Comment:
deduping is important for rate limits and for the network calls for checking if blobs exist. A lot of layers can really have an impact here - especially when pushing a bunch of tags to one repo.

Reply:
Deduping per registry is interesting... I hadn't thought of that. I think this is mostly addressed by https://github.com/google/go-containerregistry/issues/349, but I can imagine deduping blobs across multiple images to avoid extraneous existence checks (but that's not covered here).

Reply:
yep!